jumpscale/sals/reservation_chatflow/deployer.py

```diff
@@ -34,22 +34,29 @@ def __init__(self, name, workloads=None):
         self.workloads = workloads
         self.used_ips = []
         self.network_workloads = []
-        self._fill_used_ips(self.workloads)
-        self._init_network_workloads(self.workloads)
+        nodes = {node.node_id for node in j.sals.zos._explorer.nodes.list()}
+        self._fill_used_ips(self.workloads, nodes)
+        self._init_network_workloads(self.workloads, nodes)
         if self.network_workloads:
             self.iprange = self.network_workloads[0].network_iprange
         else:
             self.iprange = "can't be retrieved"
 
-    def _init_network_workloads(self, workloads):
+    def _init_network_workloads(self, workloads, nodes=None):
+        nodes = nodes or {node.node_id for node in j.sals.zos._explorer.nodes.list()}
```

Comment:
`j.sals.zos._explorer.nodes.list()` I think possible change can happen during iteration ? we can save it in a variable ?

Reply:
passed to class constructor here https://github.com/threefoldtech/js-sdk/pull/1117/commits/08674882cb7f27d00074fa5c247e5bd9a3272a56